platform/linuxbsd/detect.py

```diff
@@ -190,28 +190,30 @@ def configure(env: "Environment"):
     # FIXME: Check for existence of the libs before parsing their flags with pkg-config
 
     # freetype depends on libpng and zlib, so bundling one of them while keeping others
-    # as shared libraries leads to weird issues
-    if (
-        env["builtin_freetype"]
-        or env["builtin_libpng"]
-        or env["builtin_zlib"]
-        or env["builtin_graphite"]
-        or env["builtin_harfbuzz"]
-    ):
-        env["builtin_freetype"] = True
-        env["builtin_libpng"] = True
-        env["builtin_zlib"] = True
-        env["builtin_graphite"] = True
-        env["builtin_harfbuzz"] = True
+    # as shared libraries leads to weird issues. And graphite and harfbuzz need freetype.
+    ft_linked_deps = [
+        env["builtin_freetype"],
+        env["builtin_libpng"],
+        env["builtin_zlib"],
+        env["builtin_graphite"],
+        env["builtin_harfbuzz"],
+    ]
+    if (not all(ft_linked_deps)) and any(ft_linked_deps):  # All or nothing.
+        print(
+            "These libraries should be either all builtin, or all system provided:\n"
+            "freetype, libpng, zlib, graphite, harfbuzz.\n"
+            "Please specify `builtin_<name>=no` for all of them, or none."
+        )
+        sys.exit()
```

Comment:
This is a bit brittle, and I think the list is actually longer but with some of the components optional?
Namely fontconfig and msdfgen.